### PR TITLE
[CSM-253] [CSM-254] Generate hd wallet keys in production mode

### PR DESCRIPTION
**Pull request by Martoon-00**
This PR
1. Makes `WalletUserSecret` part of `UserSecret`
2. Puts hd wallet keys to `genesis.bin`, and uses them in production mode

New production keys (which now can be imported as wallet set) could be found here:
https://serokell.slack.com/files/martoon/F5EFFL7RR/genesis-qanet-2017-05-17.tgz

### CODE REVIEW

File: core/Pos/Crypto/SafeSigning.hs [hunk 1]
```diff
@@ -79,6 +80,10 @@ noPassEncrypt (SecretKey k) = mkEncSecret emptyPassphrase k
 checkPassMatches :: (Bi PassPhrase, Alternative f) => PassPhrase -> EncryptedSecretKey -> f ()
 checkPassMatches pp (EncryptedSecretKey _ pph) = guard (hash pp == pph)
 
+unsafeFromEncSecretKey :: Bi PassPhrase => EncryptedSecretKey -> Maybe SecretKey
```
Comment (Martoon-00) [inline core/Pos/Crypto/SafeSigning.hs]: Not used, to remove

File: src/Pos/Genesis.hs [hunk 2]
```diff
@@ -199,21 +199,27 @@ bitcoinDistributionImpl ratio coins (coinIdx, coin) =
...
         ( TxIn (unsafeHash addr) 0
         , TxOutAux (TxOut addr coin) distr
         )
     tailAddresses = map (makePubKeyAddress . fst . generateGenesisKeyPair)
         [Const.genesisN ..]
+    -- not much money to avoid making wallets slot leaders
+    hwdDistr = (mkCoin 100, [])
+    -- should be enough for testing.
+    -- greater number would increase effective number of slot leaders as well :/
+    genesisDevHdwKeyNum = 2
```
Comment (volhovM) [inline src/Pos/Genesis.hs]: Have you tested these constants with 4+ nodes using launch.sh?

Comment (Martoon-00) [inline src/Pos/Genesis.hs]: No, I'll do. What should I pay attention to?

Comment (volhovM) [inline src/Pos/Genesis.hs]: That slot leaders don't include these hdw keys adresses.

Comment (Martoon-00) [inline src/Pos/Genesis.hs]: Tried 4 and 5, everything's fine

Comment (volhovM) [inline src/Pos/Genesis.hs]: LGTM, merge.

**Comment by Martoon-00** [comment]
@gromakovsky I know you have power to merge ignoring build status.
Could you merge this one? It fails due to OOM, and contains slot leaders fix which is important to have soon.

**Pull request merged**